### PR TITLE
Migration of Mage::getResourceModel() calls

### PR DESCRIPTION
**Pull request by sshymko**
- Implemented migration of `Mage::getResourceModel()`
- Fixed migration of `Mage::getModel()->getCollection()` to resolve collections as resource models

### CODE REVIEW

File: src/Magento/Migration/Code/Processor/Mage/MageFunction/GetResourceModel.php [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+            return;
+        }
+        $this->parsed = true;
+
+        $arguments = $this->tokenHelper->getCallArguments($this->tokens, $this->index);
+
+        if ($arguments->getFirstArgument()) {
+            if ($arguments->getFirstArgument()->getFirstToken()->getType() != T_VARIABLE) {
+                $classAlias = $arguments->getFirstArgument()->getString();
+                $this->className = $this->namingHelper->getM2FactoryClassName($classAlias, 'resource_model');
```
Comment (tang-yu) [inline src/Magento/Migration/Code/Processor/Mage/MageFunction/GetResourceModel.php]: Should handle the case when resource model is obsolete and $this->className is null.

File: src/Magento/Migration/Code/Processor/Mage/MageFunction/GetResourceModel.php [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    }
+
+    /**
+     * @return string
+     */
+    protected function getModelMethod()
+    {
+        //Mage::getResourceModel('module/class')->methodName
+        $nextIndex = $this->tokenHelper->skipMethodCall($this->tokens, $this->index);
+        $nextIndex = $this->tokenHelper->getNextIndexOfTokenType($this->tokens, $nextIndex, T_STRING);
```
Comment (tang-yu) [inline src/Magento/Migration/Code/Processor/Mage/MageFunction/GetResourceModel.php]: Should handle the case where there is not method call after the getReosurceModel, e.g., $resourceModel = Mage::getResourceModel('module/class'); $resourceModel->doSomething();

File: src/Magento/Migration/Code/Processor/Mage/MageFunction/GetResourceModel.php (hunk 1 above)
Comment (sshymko) [inline src/Magento/Migration/Code/Processor/Mage/MageFunction/GetResourceModel.php]: It does, the next "if" statement.

Comment (tang-yu) [inline src/Magento/Migration/Code/Processor/Mage/MageFunction/GetResourceModel.php]: You are right, if $this->className is null, we won't do further processing anyway.

File: src/Magento/Migration/Code/Processor/Mage/MageFunction/GetResourceModel.php (hunk 2 above)
Comment (sshymko) [inline src/Magento/Migration/Code/Processor/Mage/MageFunction/GetResourceModel.php]: Copied intact from the `getModel` implementation...

Comment (tang-yu) [inline src/Magento/Migration/Code/Processor/Mage/MageFunction/GetResourceModel.php]: We do have common problem with methodName in GetModel and other classes. We can address the common problem separately.

**Pull request merged**